src/sqlfluff/rules/L041.py

```diff
@@ -93,9 +101,24 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
                 edit_segments=edit_segments,
             )
         )
-        # Delete original newlines between select keyword and select clause modifier
-        # and delete the original select clause modifier.
-        fixes.extend((LintFix.delete(s) for s in leading_newline_segments))
+
+        # Delete original newlines and whitespace between select keyword
+        # and select clause modifier.
+
+        # If there is not a newline after the select clause modifier then delete
+        # newlines between the select keyword and select clause modifier.
+        if not trailing_newline_segments:
+            fixes.extend((LintFix.delete(s) for s in leading_newline_segments))
```

Comment:
Nitpick: IIUC, the parentheses are not needed here. `LintFix.delete(s) for s in leading_newline_segments` is a generator expression, which can be passed directly to `extend()`.
```suggestion
            fixes.extend(LintFix.delete(s) for s in leading_newline_segments)
```

Reply:
Test this and you are correct. Will change and merge.